### ISSUE TITLE
devscripts: latest update fails in package()

### ISSUE THREAD

**Issue by agraef** [ItachiSan/linux_packaging#12]
Latest update fails during package():

~~~
python3 setup.py install --root="/home/ag/.cache/yay/devscripts/pkg/devscripts" --no-compile
Traceback (most recent call last):
  File "/home/ag/.cache/yay/devscripts/src/devscripts/scripts/setup.py", line 8, in <module>
    from devscripts.test import SCRIPTS
ModuleNotFoundError: No module named 'devscripts.test'
make[1]: *** [Makefile:154: install] Error 1
make[1]: Leaving directory '/home/ag/.cache/yay/devscripts/src/devscripts/scripts'
make: *** [Makefile:70: install_scripts] Error 2
==> ERROR: A failure occurred in package().
    Aborting...
 -> error making: devscripts
~~~

**Comment by ItachiSan** [comment]
I honestly do not have the issue.
Here is the asciinema of it:
[![asciicast](https://asciinema.org/a/xpfPwGH2mXISQkbsOxS9zAmwO.svg)](https://asciinema.org/a/xpfPwGH2mXISQkbsOxS9zAmwO)

**Comment by agraef** [comment]
Ok, I believe you. ;-) Must be on my side then, sorry for the false alarm.

**Comment by Ferdi265** [comment]
Hi! I can reproduce this issue and looked into it more closely, and I've found the root cause:

TL;DR: when devscripts is already installed, the PKGBUILD fails because python imports the wrong file when `devscripts.test` is imported in `devscripts/scripts/setup.py`.

Trace:
- `devscripts/scripts/setup.py` tries to import `devscripts.test`
- python looks for `devscripts`
- python does not find a `devscripts/scripts/devscripts.py`
- python does not find a `devscripts/scripts/devscripts/__init__.py`
- python finds the installed `/usr/lib/python3.10/site-packages/devscripts/__init__.py`
- python looks for `devscripts.test` in the found package
- python errors out because the `test` subpackage was not not installed with the rest

The problem is that devscripts does not have an `__init__.py` in their python package. This can be easily fixed by creating an empty `__init__.py` in `prepare()` or via `fixes.patch`.

Another workaround is to uninstall `devscripts` and reinstalling it, at least that works for me.

This also explains why @ItachiSan could not reproduce the issue: As he explains on the AUR page for this package, he does not use it himself, and therefore the build succeeds.

Fix:

```patch
diff --git a/PKGBUILD b/PKGBUILD
index a7f2dcc..506b64e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -50,6 +50,9 @@ sha256sums=('54eb392b18818460e56c052e40602bc9a1ba33c5349a58017c57b1d312b3ed9a'
 prepare(){
     cd "$pkgname"
     patch -p1 -i "$srcdir/fixes.patch"
+
+    # fix local script not being found when already installed
+    touch "$srcdir/devscripts/scripts/devscripts/__init__.py"
 }
 
 build() {
```
